java/src/main/java/com/nedap/retail/example/websocket/client/RenosApiListener.java

```diff
@@ -29,13 +29,20 @@ public void onMessage(final String message) {
             } else if (message.contains("event")) {
                 handleEvent(message);
             }
-        } catch (IOException e) {
+        } catch (final MessageParsingException e) {
+            LOG.info("The incoming message could not be parsed.");
+        } catch (final Exception e) {
             LOG.error("There was an error while handling the Renos API message", e);
         }
     }
 
-    private void handleResponse(final String message) throws IOException {
+    private void handleResponse(final String message) throws Exception {
```

Comment:
Try to avoid generating generic `Exception`s. In that case the calling side does not know what kind of exceptions can be expected. (`IOException` is in fact already quite generic)
Change to `throws IOException, MessageParsingException`